benchmarking/commons/hpo_main_common.py

```diff
@@ -36,6 +37,177 @@
 PostProcessingType = Callable[[Tuner], Any]
 
 
+def str2bool(v):
+    if isinstance(v, bool):
+        return v
+    if v.lower() in ("yes", "true", "t", "y", "1"):
+        return True
+    elif v.lower() in ("no", "false", "f", "n", "0"):
+        return False
+    else:
+        raise ArgumentTypeError("Boolean value expected.")
+
+
+@dataclass
+class Parameter:
+    name: str
+    type: Any
+    help: str
+    default: Any
+    required: bool = False
+
+
+class ConfigDict(dict):
+    """
+    Dictinary with arguments for benchmarking
+    Expected params as Parameter(name, type, default value)
+    """
+
+    __base_parameters: List[Parameter] = [
+        Parameter(
+            name="experiment_tag",
+            type=str,
+            default=None,
+            help="Tag for Syne-Tune experiments",
+        ),
+        Parameter(
+            name="num_seeds",
+            type=int,
+            default=1,
+            help="Number of seeds to run",
+        ),
+        Parameter(
+            name="start_seed",
+            type=int,
+            default=0,
+            help="First seed to run",
+        ),
+        Parameter(name="method", type=str, default=None, help="HPO method to run"),
+        Parameter(
+            name="save_tuner",
+            type=str2bool,
+            default=0,
+            help="Serialize Tuner object at the end of tuning?",
+        ),
+        Parameter(
+            name="n_workers",
+            type=int,
+            default=None,
+            help="Number of workers (overwrites default of benchmark)",
+        ),
+        Parameter(
+            name="max_wallclock_time",
+            type=int,
+            default=None,
+            help="Maximum runtime for experiment (overwrites default of benchmark)",
+        ),
+        Parameter(
+            name="random_seed",
+            type=int,
+            default=None,
+            help="Master random seed (drawn at random if not given)",
+        ),
+        Parameter(
+            name="max_size_data_for_model",
+            type=int,
+            default=None,
+            help=f"Limits number of datapoints for surrogate model of MOBSTER or HyperTune",
+        ),
+        Parameter(
+            name="scale_max_wallclock_time",
+            type=str2bool,
+            default=0,
+            help=(
+                "If 1, benchmark.max_wallclock_time is multiplied by B / min(A, B),"
+                "where A = n_workers and B = benchmark.n_workers"
+            ),
+        ),
+    ]
+
+    def __init__(self, *args, **parameters):
+        super().__init__(*args, **parameters)
+        if self.experiment_tag is None:
+            try:
+                self.experiment_tag = generate_slug(2)
+            except Exception:
+                self.experiment_tag = "syne-tune-experiment"
+
+        self.seeds = list(range(self.start_seed, self.num_seeds))
```

Comment:
It would be nice to also support `self.method_names` here, no?

Reply:
The original code has: `method_names = [args.method] if args.method is not None else list(methods.keys())`